tutorials/basic-topics/linear-search.md

```diff
@@ -0,0 +1,139 @@
+```cpp
+#include<iostream>
+using namespace std;
+
+int main(){
+  // Array to keep the values
+  int arr[5] = {1,2,3,4,5};
+  // Item to be searched 
+  int item = 3; 
+  for(int i = 0; i < 5; i++){
+    // checking if item is arr[i]
+    if(arr[i] == item){  
+      cout << "Found at " << i << endl;
+    }
+  }
+} 
+```
+
+</TabItem>
+</Tabs>
+
+### Example: [1295 - Find Numbers with Even Number of Digits](https://leetcode.com/problems/find-numbers-with-even-number-of-digits/)
+
+Here we are given an array and we are required to find out the numbers that have even number of digits.
+```
+array[] = {11,2,3456}
+Answer = 2
+Explaination = We have 11, 3456 which have even number of digits and therfore the output is 2.
+```
+We can think this problem as a linear search problem, where we are supposed to search for all the numbers that have `even` number of digits. Thats' it. Now to do so we will traverse the array and find the length of each number and check it for being even. To find the number of digits of a number, we will use some basic mathematical logic - we will take the number's log base 10 and add one to it i.e., `log10 + 1`. And if even length numbers are found then we will simply increase the count of even digits by 1. 
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@siddoinghisjob"/>
+
+```cpp
+class Solution {
+public:
+    int findNumbers(vector<int>& nums) {
+        // Initially the answer is zero
+        int answer = 0; 
+        for(int i = 0; i < nums.size(); i++){
+            // calculate the length of the number using log10 function 
+            int len = log10(nums[i])+1;
+            // check for division by 2 i.e., whether an even number
+            if(len%2 == 0) 
+                answer++;
+        }
+        return answer;
+    }
+};
+```
+
+</TabItem>
+</Tabs>
+
+### Example: [2089 - Find target Indices after sorting array](https://leetcode.com/problems/find-target-indices-after-sorting-array/)
+
+You are given a 0-indexed integer array nums and a target element target.A target index is an index i such that nums[i] == target.
+Return a list of the target indices of nums after sorting nums in non-decreasing order. If there are no target indices, return an empty list. The returned list must be sorted in increasing order.
+
+```
+Input: nums = [1,2,5,2,3], target = 2
+Output: [1,2]
+Explanation: After sorting, nums is [1,2,2,3,5].
+The indices where nums[i] == 2 are 1 and 2.
+```
+If we sort this array using library sorting functions then this problem is reduced to a simple linear search question. Where we are supposed to search for the target. Thats' it. Now to do so we will create a vetor for storing the answers, and the run a for loop and traverse every element and check it. If found then we will add that index to the vector.
+Finally we will return the vector and its' done!
+This will take **O(nlogn)** time complexity as we will be using sort function (and that takes *O(nlogn)*) time complexity.
```

Comment:
as stated in checklist, use LaTex for time complexity. no need to make it bold.

i.e.

```
$O(nlogn)$
```

which will be rendered as $O(nlogn)$